internal/helm/controller/controller.go

```diff
@@ -78,9 +80,17 @@ func Add(mgr manager.Manager, options WatchOptions) error {
 		return err
 	}
 
+	// Predicate that will filter events based on labels
+	filterPredicate, err := filterPredicate.LabelSelectorPredicate(options.Selector)
```

Comment:
What if `options.Selector` is not set?

Reply:
@estroz if no selector is specified in `watches.yaml`, then `options.Selector` becomes an empty map (i.e., `{map[] []}`).
In this case, all watched CRs get reconciled since they're no longer being "filtered" by their labels.
I also mentioned this in the issue comments [here](https://github.com/operator-framework/operator-sdk/issues/4949#issuecomment-865253483).

Reply:
So now you're running an empty, extraneous predicate on each event retrieved from the apiserver. While what you say is true, I'd rather see this added conditionally on `options.Selector != nil`.

Reply:
Okay, I'll add a check for an empty selector. 
`options.Selector` is a `metav1.LabelSelector` type and since it comes in as an empty map, the `!= nil` check is not compatible. The workaround I could come up with was doing a `DeepEquals()` with an empty selector. That seems to do the comparison correctly.